### ISSUE TITLE
Duplicate query results

### ISSUE THREAD

**Issue by mrsdpz** [RedisJSON/RedisJSON#667]
`json.get student`

```json
[{"name":{"first":"Markss","middle":"S","last":"Pronto"},"rank":1},{"name":{"first":"A","middle":"A","last":"Pronto"},"rank":8},{"name":{"first":"A","middle":"A","last":"Pronto"},"rank":90}]
```

`json.get student '$.[?(@.name.first=="A")]'`

```json
[{"name":{"first":"A","middle":"A","last":"Pronto"},"rank":8},{"name":{"first":"A","middle":"A","last":"Pronto"},"rank":8},{"name":{"first":"A","middle":"A","last":"Pronto"},"rank":90},{"name":{"first":"A","middle":"A","last":"Pronto"},"rank":90}]
```

**Comment by gkorland** [comment]
@mrsdpz a fix was pushed see https://github.com/RedisJSON/jsonpath/pull/29
Can you please update and retest?

**Comment by oshadmi** [comment]
@mrsdpz @gkorland Since RedisJSON is using `Cargo.lock`, the fix is blocked by #691
